modules/react/package.json

```diff
@@ -36,9 +36,12 @@
   "scripts": {},
   "peerDependencies": {
     "@deck.gl/core": "^9.0.0-beta",
-    "@types/react": "^18.2.0",
     "react": "^18.2.0",
     "react-dom": "^18.2.0"
   },
+  "devDependencies": {
+    "@types/react": "^18.2.0",
+    "@types/react-dom": "^18.2.0"
+  },
```

Comment:
If any part of our public API exposes these types, I believe they should be in either `dependencies` or `peerDependencies`. Example:

https://github.com/visgl/deck.gl/blob/89189f4e9ecb2f1b9b619f9a66c246690948fd19/examples/layer-browser/src/components/layer-controls.jsx#L38

Unfortunately, that does mean installing types is imposed on non-typescript users, but those packages should be small.

Reference:

- https://stackoverflow.com/a/46011417

Reply:
My reasoning is this
- If someone uses typescript with React, then they must already have the react types installed
- for someone not using typescript, no lib check will be performed, and all .d.ts files in the dist directory will be ignored anyways

If react is in peer dep then its types must be in peer dep too, as their versions need to match. (I just noticed that we are using `^`, it should be `>=`) Which means non-typescript users have to manually add the `@types` packages to their dependency.

Reply:
If we wanted to play it very safe on this one, I think the approach would be...

dependencies: 
- `"@types/react": "*"`

peerDependencies: 
- `"react": ">=18.2.0"`

devDependencies: 
- `"@types/react": "^18.2.0"`
- `"react": "^18.2.0"`

That said, your observation that anyone using TS+React would have these installed already makes sense, and I'm probably overcomplicating it. No objections to your change!

Reply:
I have reverted a previous change so the peer dependency version is now `"react": ">=16.3.0` (earliest version of the hooks API). Assume we add `"@types/react": ">=16.3.0"` to dependencies and a React 16 application imports it:

```json
"dependencies": {
  "@types/react": "^16.9.0",
  "react": "^16.9.0",
  "deck.gl": "^9.0.0"
}
```

depending on the package manager and order of operation, they may end up with something like:

```
├─ react@16.9.0
├─ @types/react@16.9.0
├─ @deck.gl/react@9.0.0
│  └─ @types/react@18.2.0
```

Reply:
I don't know how to prevent that, let's go with your approach. 👍🏻